code/modules/clothing/masks/transmog.dm

```diff
@@ -43,6 +43,10 @@
 					C.transmogrify(target_type)
 				else
 					C.transmogrify(target_type, revert_spell_type || TRUE)
+				var/turf/T = get_turf(C)
```

Comment:
Put it above the transmogrifies so it gives the key_name properly, maybe?

Reply:
Yes this is a good idea